### PR TITLE
Add access to the methods `Node::evaluate`  to each custom Op which inherit from ov::op::Op

### PR DESCRIPTION
**Pull request by artkuli**
### Tickets:
 - *82943*

### CODE REVIEW

File: src/core/include/openvino/op/op.hpp [hunk 1]
```diff
@@ -14,12 +14,13 @@
...
-                                                             _OPENVINO_RTTI_WITH_TYPE_VERSION_PARENT,  \
-                                                             _OPENVINO_RTTI_OP_WITH_TYPE_VERSION,      \
-                                                             _OPENVINO_RTTI_OP_WITH_TYPE)(__VA_ARGS__))
+#define OPENVINO_OP(...)                                                                                \
+    _OPENVINO_RTTI_EXPAND(_OPENVINO_RTTI_DEFINITION_SELECTOR(__VA_ARGS__,                               \
+                                                             _OPENVINO_RTTI_WITH_TYPE_VERSIONS_PARENT,  \
+                                                             _OPENVINO_RTTI_WITH_TYPE_VERSION_PARENT,   \
+                                                             _OPENVINO_RTTI_OP_WITH_TYPE_VERSION,       \
+                                                             _OPENVINO_RTTI_OP_WITH_TYPE)(__VA_ARGS__)) \
+    using ov::op::Op::evaluate;
```
Comment (itikhono) [inline src/core/include/openvino/op/op.hpp:23]: can we add a comment here?

Comment (artkuli) [reply src/core/include/openvino/op/op.hpp:24]: Added

File: src/core/tests/op_eval/swish.cpp [hunk 2]
```diff
@@ -75,3 +75,14 @@ TEST(op_eval, swish_without_beta) {
     for (size_t i = 0; i < inputs.size(); i++)
         EXPECT_NEAR(result_data[i], expected_result[i], 0.000001);
 }
+
+TEST(op_eval, swish_new_evaluate) {
+    Shape shape{3};
+    auto p = make_shared<op::Parameter>(element::f32, shape);
+    auto beta = make_shared<op::Parameter>(element::f32, Shape{});
+    auto swish = make_shared<op::v4::Swish>(p, beta);
```
[the remark this answers is not in the document]
Comment (artkuli) [reply src/core/tests/op_eval/swish.cpp]: Fixed

File: src/core/include/openvino/op/op.hpp (hunk 1 above)
Comment (itikhono) [reply src/core/include/openvino/op/op.hpp:24]: it's better to add more details, it's hard to understand what is the root cause of the issue, usually c++ allows to use methods of Base class. But if they have the same name and one of them is overrided in Derived class, it doesn't.
https://onlinegdb.com/i-QNi4uCj

Comment (artkuli) [reply src/core/include/openvino/op/op.hpp:28]: Done

**Review by itikhono — APPROVED** [review]
LGTM

**Review by ilyachur — APPROVED** [review]

**Pull request merged**